### PR TITLE
ycsb: add JSONB mode

### PR DESCRIPTION
**Pull request by justinj**
Addresses #112.

This commit adds the -json flag, which, rather than storing the 10
fields in separate columns, stores them in one large JSONB object.

Feedback welcome on whether the translation performed here is a fair
one.

Will report back with numbers from a GCE worker, but my laptop shows the following:

```
~/g/s/g/c/loadgen (add-jsonb|✔) $ ./ycsb/ycsb -drop
elapsed______ops/sec__reads/empty/errors___writes/errors____scans/errors
     1s       6899.6           0 / 0 / 0        6938 / 0           0 / 0
Loading complete: 1.5s
     1s       2629.3        9077 / 0 / 0 18446744073709545162 / 0           0 / 0
     2s      18551.5       17595 / 0 / 0         925 / 0           0 / 0
     3s      18922.2       18022 / 0 / 0         913 / 0           0 / 0
     4s      18992.2       18003 / 0 / 0         978 / 0           0 / 0
     5s      18680.3       17806 / 0 / 0         886 / 0           0 / 0
     6s      18763.8       17795 / 0 / 0         973 / 0           0 / 0
     7s      18737.1       17808 / 0 / 0         912 / 0           0 / 0
     8s      19179.9       18261 / 0 / 0         968 / 0           0 / 0
     9s      18896.1       17954 / 0 / 0         911 / 0           0 / 0
    10s      18782.8       17799 / 0 / 0         970 / 0           0 / 0
^C
elapsed__ops/sec(total)__errors(total)
   9.8s        18802.0              0
~/g/s/g/c/loadgen (add-jsonb|✔) $ ./ycsb/ycsb -drop -json
elapsed______ops/sec__reads/empty/errors___writes/errors____scans/errors
     1s       6659.2           0 / 0 / 0        6699 / 0           0 / 0
Loading complete: 1.6s
     0s        396.6        6724 / 0 / 0 18446744073709545287 / 0           0 / 0
     1s      18320.1       17411 / 0 / 0         914 / 0           0 / 0
     2s      19063.6       18115 / 0 / 0         965 / 0           0 / 0
     3s      18013.6       17091 / 0 / 0         898 / 0           0 / 0
     4s      18700.7       17736 / 0 / 0         986 / 0           0 / 0
     5s      19334.0       18426 / 0 / 0         938 / 0           0 / 0
     6s      18708.5       17761 / 0 / 0         932 / 0           0 / 0
     7s      19114.0       18147 / 0 / 0         945 / 0           0 / 0
     8s      18870.3       17871 / 0 / 0         986 / 0           0 / 0
     9s      18368.8       17479 / 0 / 0         934 / 0           0 / 0
    10s      18742.2       17804 / 0 / 0         908 / 0           0 / 0
^C
elapsed__ops/sec(total)__errors(total)
  11.0s        18672.1              0
```

The numbers seem a little inconsistent though, so hopefully running on a GCE worker (or roachperf?) will give a clearer image.

### CODE REVIEW

File: ycsb/main.go [hunk 1]
```diff
@@ -342,6 +344,77 @@ type cockroach struct {
 	writeStmt *sql.Stmt
 }
 
+func initRelationalStrategy(db *sql.DB) (*sql.Stmt, *sql.Stmt, error) {
```
Comment (rjnn) [inline ycsb/main.go]: comment needed, to explain what the three returned values are.

**Review by rjnn — APPROVED** [review]
LGTM. I think you've overengineered this thing, but its good, so merge it! :)

File: ycsb/main.go (hunk 1 above)
Comment (rjnn) [reply ycsb/main.go]: actually just rip out this complexity.

im not the biggest fan of this statefulness, where it executes the create statement and then returns the read and write statements. why not just have three statically defined arrays of strings?
`createStmt[schema]`, `readStmt[schema]` ...
then a simple `schema = JSON` means does all the work for you. then you dont even need these `strategy` helper methods.

Anyway I'm not going to insist on this, but food for future thought.

**Comment by justinj** [comment]
Oh yes you're totally right that that was overkill. PTAL.

File: ycsb/main.go [hunk 2]
```diff
@@ -416,22 +462,24 @@ func setupCockroach(dbURLs []string) (database, error) {
...
-	FIELD7 TEXT,
-	FIELD8 TEXT,
-	FIELD9 TEXT,
-	FIELD10 TEXT
-)`
-	if _, err := db.Exec(createStmt); err != nil {
+	var schema [3]string
+	if *json {
+		schema = jsonbStrategy
+	} else {
```
Comment (rjnn) [inline ycsb/main.go]: you can elide the else, as is apparently the fashion in the linted repos.
```
schema := relationalStrategy
if *json {
   schema = jsonbStrategy
}
```

**Review by rjnn — APPROVED** [review]
LGTM :shipit:

**Pull request merged**